activemodel/lib/active_model/attribute_methods/before_type_cast.rb

```diff
@@ -1,10 +1,10 @@
 # frozen_string_literal: true
 
-module ActiveRecord
+module ActiveModel
   module AttributeMethods
-    # = Active Record Attribute Methods Before Type Cast
+    # = Active Model Attribute Methods Before Type Cast
     #
-    # ActiveRecord::AttributeMethods::BeforeTypeCast provides a way to
+    # ActiveModel::AttributeMethods::BeforeTypeCast provides a way to
     # read the value of the attributes before typecasting and deserialization.
     #
     #   class Task < ActiveRecord::Base
```

Comment:
`ActiveRecord::AttributeMethods::BeforeTypeCast` is a public module, so I don't think we can remove it without a deprecation.

I think this is also worth splitting out into a separate PR.  I actually made a while ago: #44534.  I've rebased it and added you as a co-author.

Reply:
Great! Thanks @jonathanhefner